cluster/log-dump/log-dump.sh

```diff
@@ -312,8 +329,25 @@ function dump_nodes_with_logexporter() {
     return
   fi
 
-  # Give some time for the pods to finish uploading logs.
-  sleep "${logexport_sleep_seconds}"
+  # Periodically fetch list of already logexported nodes to verify
+  # if we aren't already done.
+  start="$(date +%s)"
+  while true; do
+    now="$(date +%s)"
+    if [[ $((now - start)) -gt ${logexport_sleep_seconds} ]]; then
+      echo "Waiting for all nodes to be logexported timed out."
+      break
+    fi
+    if nodes_succeeded=$(gsutil ls ${gcs_artifacts_dir}/logexported-nodes-registry); then
+      echo "Successfully listed marker files for successful nodes"
+      LOGEXPORTED_NODES=${nodes_succeeded}
```

Comment:
Why not set `LOGEXPORTED_NODES` directly above instead of another variable?

Reply:
done